src/utils/components/SidebarEditor/SidebarEditor.tsx

```diff
@@ -0,0 +1,41 @@
+import * as React from 'react';
+import { dump } from 'js-yaml';
+
+import { YAMLEditor } from '@openshift-console/dynamic-plugin-sdk';
+import { Sidebar, SidebarContent, SidebarPanel } from '@patternfly/react-core';
+
+import Loading from '../Loading/Loading';
+
+import { SidebarEditorContext } from './SidebarEditorContext';
+
+import './sidebar-editor.scss';
+
+type SidebarEditorProps = {
+  resource:
+    | string
+    | {
+        [key: string]: any;
+      };
+};
+
+const SidebarEditor: React.FC<SidebarEditorProps> = ({ children, resource }) => {
+  const { showEditor } = React.useContext(SidebarEditorContext);
+  return (
+    <Sidebar isPanelRight hasGutter hasNoBackground className="sidebar-editor">
+      <SidebarContent>{children}</SidebarContent>
+      {showEditor && (
+        <SidebarPanel
+          width={{ default: 'width_33', lg: 'width_50', xl: 'width_50' }}
+          style={{ minHeight: '100%' }}
+          className="sidebar-editor"
+        >
+          <React.Suspense fallback={<Loading />}>
+            <YAMLEditor value={dump(resource)} minHeight="300px" />
```

Comment:
dump may result in exception, should be called outside with try catch block

Reply:
we'll use dump with `skipInvalid` option